src/ipc/ipc4/handler.c

```diff
@@ -38,8 +38,15 @@
 struct ipc4_msg_data {
 	uint32_t msg_in[2]; /* local copy of current message from host header */
 	uint32_t msg_out[2]; /* local copy of current message to host header */
+	bool delay_reply;
+	uint32_t delay_error;
```

Comment:
Why do we need both delay_reply and delay_error ? i.e. if delay_error != 0 then we have an error ?

Reply:
delay_error is for error type, since a few message will be delayed with different error message type.  If only start stream is delayed, we can remove this delay_error and just return pipeline error for ipc4